pkg/services/ngalert/provisioning/notification_policies.go

```diff
@@ -131,6 +131,10 @@ func (nps *NotificationPolicyService) ResetPolicyTree(ctx context.Context, orgID
 	if err != nil {
 		return definitions.Route{}, err
 	}
+	// When resetting the policy, we have to make sure to use a receiver that
+	// does exist. Otherwise it can cause everything to crash as the default
+	// receiver used could already be deleted by the user.
+	route.Receiver = revision.cfg.AlertmanagerConfig.Receivers[0].Name
```

Comment:
What if `len(Receivers)` is `0`?

Reply:
It can never be zero, you always have at least one contact point.

Reply:
We see configs get broken all the time.. and a future API bug could also break this assumption. Should we have a step to add the default contact point if it's zero anyways?

Even if it never gets hit, adding the check makes it obvious to the reader that this case is handled, the behavior is not so implicit and easier to understand at a glance.